examples/audio/librispeech.py

```diff
@@ -0,0 +1,98 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+import os
+
+from pathlib import Path
+from typing import Union
+
+import torchaudio
+
+from torchdata.datapipes.iter import FileOpener, HttpReader, IterableWrapper
+
+
+URL = "train-clean-100"
+FOLDER_IN_ARCHIVE = "LibriSpeech"
+BASE_URL = "http://www.openslr.org/resources/12/"
+_CHECKSUMS = {
+    "dev-clean.tar.gz": "76f87d090650617fca0cac8f88b9416e0ebf80350acb97b343a85fa903728ab3",
+    "dev-other.tar.gz": "12661c48e8c3fe1de2c1caa4c3e135193bfb1811584f11f569dd12645aa84365",
+    "test-clean.tar.gz": "39fde525e59672dc6d1551919b1478f724438a95aa55f874b576be21967e6c23",
+    "test-other.tar.gz": "d09c181bba5cf717b3dee7d4d592af11a3ee3a09e08ae025c5506f6ebe961c29",
+    "train-clean-100.tar.gz": "d4ddd1d5a6ab303066f14971d768ee43278a5f2a0aa43dc716b0e64ecbbbf6e2",
+    "train-clean-360.tar.gz": "146a56496217e96c14334a160df97fffedd6e0a04e66b9c5af0d40be3c792ecf",
+    "train-other-500.tar.gz": "ddb22f27f96ec163645d53215559df6aa36515f26e01dd70798188350adcb6d2",
+}
+AUDIO_EXT = ".flac"
+TXT_EXT = ".trans.txt"
+
+
+def load_librispeech_item(audio_file):
+    audio_filename = os.path.basename(audio_file)
+    speaker_id, chapter_id, utterance_id = audio_filename.split("-")
+    path = os.path.dirname(audio_file)
+
+    text_file = os.path.join(path, speaker_id + "-" + chapter_id + TXT_EXT)
+
+    # Load audio
+    waveform, sample_rate = torchaudio.load(audio_file)
+
+    # Load text
+    with open(text_file) as ft:
+        for line in ft:
+            fileid_text, transcript = line.strip().split(" ", 1)
+            if audio_filename == fileid_text:
+                break
+        else:
+            # Translation not found
+            raise FileNotFoundError("Translation not found for " + fileid_audio)
+
+    return (
+        waveform,
+        sample_rate,
+        transcript,
+        int(speaker_id),
+        int(chapter_id),
+        int(utterance_id),
+    )
+
+
+def LibriSpeech(root: Union[str, Path], url: str = URL, folder_in_archive: str = FOLDER_IN_ARCHIVE):
```

Comment:
should it be a datapipe class instead of a method?

Reply:
We try to use modular DataPipe to build pipeline (a sequence of DataPipe). So, I am using a function to represent how you want to create the pipeline.
The way you build your Dataset is up to domain. For example, TorchText is using method to return DataPipe https://github.com/pytorch/text/tree/main/torchtext/datasets. But, for TorchVision, they create another Wrapper https://github.com/pytorch/vision/tree/main/torchvision/prototype/datasets/_builtin and create a common API to fetch data pipeline `_make_datapipe`